### PR TITLE
Added unit tests

### PR DESCRIPTION
**Pull request by SevlaMare**
-Initial Setup
-Added test cases
-Linter fixes

### Brief about test cases
* my_each is tested against:
Numbers, strings, range, without block and edge case with an array with single nil element, as well return enumerable if no block is given.

* my_each_with_index
Loop through index and element on array of numbers and string, return enumerable if no block is given.

* my_select
Filter array based on criteria, return enumerable if no block is given.

* my_map
Can map arrays of numbers and strings, accept procs as argument and return enumerable if no block is given.

* my_count
Count elements of array if no args neither block is given, count how many times the arg appear on the array, count elements given a condition as block.

* my_inject
Cumulative operation with operators plus, minus and multiply, as well with a given block condition.

* my_all
Return true/false after match de conditions, tested with empty array, containing nil, using blocks and regex.

* my_any
Runs with no blocks, work with empty array, recognize different classes of numbers, accept blocks and regex.

* my_none
Runs with no blocks, work with empty array, recognize different classes of numbers, accept blocks and regex.

* multiply_els
Test my_inject with expected behavior.

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @SevlaMare :clap: However, some changes need to be made in order for us to approve your project. Check :arrow_down: 

- **[REQUIRED]** Add info about tests, instructions about how to run them and what you are testing. :mag: 

- **[REQUIRED]** The spec folder is missing the 'spec_helper.rb' file, it prevents me from running the tests.
- **[REQUIRED]** For users who don't have **rspec** installed globally, they would need a 'Gemfile' to run your tests. It's better to take their setup into consideration.

### That's it :relieved: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by SevlaMare** [comment]
* Added information about all tests in the pull request.
* Added spec_helper.rb file
* Added instructions to install and run RSpec

**Comment by SevlaMare** [comment]
* Added information about all tests in the pull request.
* Added spec_helper.rb file
* Added instructions to install and run RSpec

File: spec/main_spec.rb [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    end
+
+    it 'iterates through a range' do
+      expect(range_forward.my_each { |x| x })
+        .to eql(range_forward.to_a)
+    end
+
+    it 'If no block is given, an enumerator is returned.' do
+      expect(array_numeric.my_each)
+        .to be_a(Enumerable)
```
Comment (shshamim63) [inline spec/main_spec.rb]: To be specific it returns an Enumerator.

**Review by shshamim63 — APPROVED** [review]
## Status :100: Approved :100: 
___
Hi @SevlaMare :wave: ,
Nice job :clap: , You have covered all the changes asked by the previous reviewer. I have a suggestion to make. And it is added in the comment below. So nothing is blocking me now to approve this project.
___
##### This :+1: is from me and you can go ahead. 
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
